internal/errorlint/errorlint.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one file package.
+
+// Package errorlint contains the necessary logic for the error/log/trace linter.
+package errorlint
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/getoutreach/lintroller/internal/common"
+	"github.com/getoutreach/lintroller/internal/reporter"
+	"golang.org/x/tools/go/analysis"
+)
+
+// name defines the name for the errorlint linter.
+const name = "errorlint"
+
+// doc defines the help text for the error linter.
+const doc = `Ensures that each static message in error/trace/log/ is lowercase.
+
+A valid example is the following:
+
+	func foo() { errors.New("org not found in launchdarkly rule")}`
+
+// Analyzer exports the errorlint analyzer (linter).
+var Analyzer = analysis.Analyzer{
+	Name: name,
+	Doc:  doc,
+	Run:  errorlint,
+}
+
+// errorlint defines linter for error/trace/log messages
+func errorlint(_pass *analysis.Pass) (interface{}, error) {
+	// Ignore test packages.
+	if common.IsTestPackage(_pass) {
+		return nil, nil
+	}
+
+	// Wrap _pass with reporter.Pass to take nolint directives into account.
+	pass := reporter.NewPass(name, _pass, reporter.Warn())
+	for _, file := range pass.Files {
+		// Ignore generated files and test files.
+		if common.IsGenerated(file) || common.IsTestFile(pass.Pass, file) {
+			continue
+		}
+	}
```

Comment:
This loop is a no-op. I'd need to pull down your branch to provide a suggestion on how to refactor this because I don't have all of this analysis stuff committed to memory anymore, but what you want to be doing at a high level here is iterating over individual files, skipping generated and test files, and iterating over the symbols (that you can about) in those non-generated and non-test files.

This loop is iterating over files and not actually doing anything. All of your meaningful work appears to be in the loop below which most likely will trigger lint warnings on generated and test files. So that needs to be addressed. I can help out if you need it 🙂

Reply:
I'll try that and take this offline with you, as I couldn't find a way to get typesInfo from file. If I just use pass it repeats the error.

Reply:
I was able to figure it out using node walker for file